components/server/src/user/user-service.ts

```diff
@@ -77,12 +79,19 @@ export class UserService {
     }
 
     async findUserById(userId: string, id: string): Promise<User> {
-        await this.authorizer.checkPermissionOnUser(userId, "read_info", id);
+        if (userId !== id) {
```

Comment:
Why is this necessary? :thinking:

Reply:
to detangle a chicken&egg problem we have here: the user might not be migrated when this is called the first time with the feature flag on, because that only happens below

Reply:
Thanks!